### PR TITLE
feat: [IA-945] Add a screen to preferences to opt-in/out reminders notifications

### PR DESCRIPTION
**Pull request by emiliopavia**
### ⚠️ **This PR depends on #4126**
This PR implements a new preference screen, in order to control the new `reminder_status` property of user's profile. The new feature is behind feature flag, and is still disabled in production.

**_Please note_** that the switch to control **notification previews** is still **missing** here, and will be included in another PR. Also **english** localization is **missing** here.

| List | Detail | Error
| - | - | - |
| Simulator Screen Shot - iPhone 13 - 2022-10-06 at 09 38 11 | Simulator Screen Shot - iPhone 13 - 2022-10-06 at 09 38 19 | Simulator Screen Shot - iPhone 13 - 2022-10-06 at 09 51 41 |

## List of changes proposed in this pull request
- Updated `io-backend` specs
- `BasePaymentFeatureListItem` has been renamed to `PreferencesListItem` (and moved to `ts/components`) in order to be reused in the new screen
- Added `profileRemindersPreferenceSelector`
- Updated reducers to store the `reminder_status` from `/profile` responses
- Created `NotificationsPreferencesScreen` (and added to the profile navigator)
- Added some component tests for the new screen

## How to test
Use [this](https://github.com/pagopa/io-dev-api-server/tree/IA-948-add-reminder-status) branch of the `dev-server` and set the `REMINDERS_OPT_IN_ENABLED` feature flag to `YES`. Navigate to the new screen and toggle the preference.

### CODE REVIEW

**Comment by thisisjp** [comment]
@AliceAzzolini will provide the english strings by EOD

File: locales/en/index.yml [hunk 1]
```diff
@@ -499,6 +502,14 @@ profile:
       contextualHelpContent: !include profile/profile_preferences_calendar.md
       local_calendar: Local calendar
       birthday_calendar: Birthdays
+    notifications:
+      header: Notifiche
+      title: Personalizza le notifiche
```
Comment (AliceAzzolini) [inline locales/en/index.yml:507]:
```suggestion
      title: Customize push notifications
```

File: locales/en/index.yml [hunk 2]
```diff
@@ -499,6 +502,14 @@ profile:
...
+      subtitle: Scegli come ricevere aggiornamenti sui messaggi e promemoria che ricevi.
+      reminders:
+        title: Consenti i promemoria
+        description: Ricevi notifiche in prossimità di scadenze o quando non hai ancora letto dei messaggi
```
Comment (AliceAzzolini) [inline locales/en/index.yml:511]:
```suggestion
        description: Receive push notifications close to deadlines or when you have unread messages
```

File: locales/it/index.yml [hunk 3]
```diff
@@ -499,6 +502,14 @@ profile:
       contextualHelpContent: !include profile/profile_preferences_calendar.md
       local_calendar: Calendario locale
       birthday_calendar: Compleanni
+    notifications:
+      header: Notifiche
+      title: Personalizza le notifiche
```
Comment (AliceAzzolini) [inline locales/it/index.yml:507]:
```suggestion
      title: Personalizza le notifiche push
```

File: locales/it/index.yml [hunk 4]
```diff
@@ -499,6 +502,14 @@ profile:
...
+      subtitle: Scegli come ricevere aggiornamenti sui messaggi e promemoria che ricevi.
+      reminders:
+        title: Consenti i promemoria
+        description: Ricevi notifiche in prossimità di scadenze o quando non hai ancora letto dei messaggi
```
Comment (AliceAzzolini) [inline locales/it/index.yml:511]:
```suggestion
        description: Ricevi notifiche push in prossimità di scadenze o quando non hai ancora letto dei messaggi
```

**Review by thisisjp — APPROVED** [review]

File: locales/en/index.yml [hunk 5]
```diff
@@ -483,6 +483,9 @@ profile:
           title: Do you want to change the language to
           subtitle: You'll need to restart the application to make this happen
         not_selected: Not selected
+      notifications:
+        title: Push notifications
+        subtitle: Consenti anteprima e promemoria
```
Comment (CrisTofani) [inline locales/en/index.yml:488]: @thisisjp @AliceAzzolini the English version of this key is missing by purpose or would you like to suggest a translation for the key? 😄

**Review by CrisTofani — CHANGES_REQUESTED** [review]
PR LGTM, just a single mention for the locale translation as highlihted

File: locales/en/index.yml (hunk 5 above)
Comment (AliceAzzolini) [reply locales/en/index.yml:488]: @CrisTofani that was not intended, here is the translation:

```suggestion
        subtitle: Allow previews and reminders

**Review by CrisTofani — APPROVED** [review]

**Pull request merged**